src/main/java/org/sagebionetworks/web/client/Portal.java

```diff
@@ -57,6 +57,13 @@ public void onModuleLoad() {
     if (initToken.length() > 0 && initToken.startsWith("!")) {
       String fullUrl = Window.Location.getHref();
       fullUrl = fullUrl.replace("#!", "");
+      // SWC-6854: Also detect a tables query in the hash fragment.  If exist, move just the query to the hash fragment
+      if (
+        initToken.startsWith("!Synapse") && initToken.contains("/tables/query/")
+      ) {
+        fullUrl = fullUrl.replace("/tables/query/", "/tables/#query/");
```

Comment:
For old links, actively move the query portion to the hash fragment during client-side app init.